src/sidebar/services/frame-sync.js

```diff
@@ -366,9 +366,23 @@ export class FrameSyncService {
 
     guestRPC.on(
       'showAnnotations',
-      /** @param {string[]} tags */ tags => {
-        this._store.selectAnnotations(this._store.findIDsForTags(tags));
+      /**
+       * @param {string[]} tags
+       * @param {boolean} focusFirstInSelection - Whether to give keyboard focus
+       *   to the first annotation in the selection.
+       */
+      (tags, focusFirstInSelection = false) => {
+        const ids = this._store.findIDsForTags(tags);
+        this._store.selectAnnotations(ids);
         this._store.selectTab('annotation');
+
+        // Attempt to transfer keyboard focus to the first selected annotation.
+        // This may be blocked in WebKit-based browsers if the user has not
+        // previously interacted with the frame.
+        if (ids.length > 0 && focusFirstInSelection) {
```

Comment:
Note that the first `id` in `ids` isn't guaranteed to be the first `id` that was provided to this callback. Do we care?

Reply:
Are you referring to the fact that IDs won't be present in the list if an annotation with a given tag doesn't have an ID? Otherwise the order is preserved. I think that's OK as a limitation. It just means that both selecting and focusing annotations only works with saved annotations at present. If you click a highlight corresponding to an unsaved annotation you'll see that the client does not create a selection. It would be nice to remove that limitation, but I think that is out of scope for this PR.